Add Git Crash Course

nbhuiyan opened the pull request:
This documentation is meant to help potential contributors who
are not used to working with git learn what is necessary
to contribute to Eclipse OMR.

[skip ci]

Signed-off-by: Nazim Uddin Bhuiyan <nazimudd@ualberta.ca>

0xdaryl commented on CONTRIBUTING.md:
```diff
@@ -28,6 +28,9 @@ We welcome and encourage all kinds of contribution to the project, not only code
 This includes bug reports, user experience feedback, assistance in reproducing
 issues and more.
 
```
Can you update the copyright to 2020 please?

andrewcraik commented on doc/GitCrashCourse.md:
```diff
@@ -0,0 +1,543 @@
+# Git crash course
+
+Git is an open-source distributed version control system that has quickly
+become the most popular version control system for tracking code changes
```
this feels like something that could become stale/controversial. It doesn't really add anything - eg why do we care it is so popular - it is what this project uses at this time...

andrewcraik commented on doc/GitCrashCourse.md:
```diff
@@ -0,0 +1,543 @@
+
+# Setting up git and GitHub
+
+Although there are many git GUIs available for download, this guide will
```
This doesn't seem to add much - why not just say this guide will focus on using the git cli - you may choose to use a GUI in addition / instead if you so choose.

andrewcraik commented:
I would suggest in the setup noting that the email used in the global config should match the email used to sign the ECA and the fact that one needs to sign the ECA before a contribution can be accepted by the project (even if this is just a link to another one of our documents) since ECA issues are one of the more common PR issues for new contributors.

nbhuiyan commented:
@andrewcraik Thanks for the feedback! I have addressed your comments.

Regarding setting the global email setting to the email used to sign the ECA, there's also the option of setting a OMR project specific configuration to override with the ECA-associated email instead. I mentioned that and also pointed to the Legal considerations section of CONTRIBUTING.md.

0xdaryl commented on doc/GitCrashCourse.md:
```diff
@@ -0,0 +1,552 @@
+page](https://git-scm.com/download/win), and following the installation
+wizard.
+
+---
```
I suggest adding something like the following here:

```
### Sign the Eclipse Contributor Agreement (ECA)

Before you can make any code contributions to the public GitHub repository you must
sign the Eclipse Contributor Agreement (ECA).  This is a simple process outlined in the
[Contribution Guidelines](../CONTRIBUTING.md#legal-considerations).  This step is
important because each of your commits must be signed off by the same email you
used to sign the ECA.

---
```

0xdaryl commented on doc/GitCrashCourse.md:
```diff
@@ -0,0 +1,560 @@
+would just like some feedback as you continue adding more commits, simply
+prefix your Pull Request title with `WIP` (ie, Work In Progress), and/or
+open the Pull Request as a
+[Draf Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/).
```
Apologies for missing this earlier.   "Draft" is spelled incorrectly.

0xdaryl commented:
@andrewcraik , have your concerns been addressed?
